src/main/java/org/folio/linked/data/integration/kafka/sender/inventory/InstanceUpdateMessageSender.java

```diff
@@ -0,0 +1,41 @@
+package org.folio.linked.data.integration.kafka.sender.inventory;
+
+import static org.folio.linked.data.util.BibframeUtils.extractInstances;
+import static org.folio.linked.data.util.Constants.FOLIO_PROFILE;
+
+import java.util.Collection;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.folio.linked.data.integration.kafka.sender.UpdateMessageSender;
+import org.folio.linked.data.mapper.kafka.inventory.KafkaInventoryMessageMapper;
+import org.folio.linked.data.model.entity.Resource;
+import org.folio.search.domain.dto.InstanceIngressEvent;
+import org.folio.search.domain.dto.InstanceIngressEvent.EventTypeEnum;
+import org.folio.spring.tools.kafka.FolioMessageProducer;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+@Service
+@Profile(FOLIO_PROFILE)
+@RequiredArgsConstructor
+public class InstanceUpdateMessageSender implements UpdateMessageSender {
+
+  private final KafkaInventoryMessageMapper kafkaInventoryMessageMapper;
+  private final FolioMessageProducer<InstanceIngressEvent> instanceIngressMessageProducer;
+
+  @Override
+  public Collection<Resource> apply(Resource resource) {
+    return extractInstances(resource)
```

Comment:
Can we add a validation in `extractInstances` method just to be safe?


  public static List<Resource> extractInstances(Resource resource) {
    if (resource.isOfType(INSTANCE)) {
      return List.of(resource);
    }
    if (resource.isOfType(WORK)) {  -> Add this new validation
      return resource.getIncomingEdges().stream()
        .filter(re -> INSTANTIATES.getUri().equals(re.getPredicate().getUri()))
        .map(resourceEdge -> {
          var instance = resourceEdge.getTarget();
          instance.addOutgoingEdge(resourceEdge);
          return instance;
        })
        .toList();
  }
  return List.of();
}

Reply:
We anyway searching for the INSTANTIATES predicate, which is a filter by itself - it could belong to a Work resource only, right? Do you see it differently?